### DISCUSSION TITLE
Generate Basemap z0-z14

### DISCUSSION THREAD

**Discussion by BenOnTrack** [systemed/tilemaker#531] [Q&A]
Hi, I have a similar question as https://github.com/systemed/tilemaker/discussions/480 but in my case I'd like to generate:
- the entire world from z0-z14 - to serve as a **basemap** (only countries border, coastline and ocean)
- that would be served in parallel of a specific area, containing higher level of details (POI, landuse, landcover, roads, ...)

In a way, that would be a similar request to what openmaptiles is proposing (I've noticed that in their process they seem to import a lot of shapefiles from Natural Earth Data). 
I'd like to stick to Tilemaker since I find it way easier to customize than openmaptiles.

### Part 1 : Generation

I kind of manage to generate a similar basemap with tilemaker, but i can't make it fit in a small mbtiles (openmaptiles for Basemap+Berlin fits in less than 13mb while my Basemap is already over 30mb for only zo-z7...).

I've been generating this basemap with --bbox -180,-90,180,90 (and without providing any .pbf file)

As anyone ever tried to generate such a basemap ? and if so, if you could share your config file

At the moment, I'm using the following one:
There might be something smart to do with the `simplify_level `or `simplify_ratio` ?
Also, according to the process run by openmaptiles, I'm pretty sure I'm missing a bunch of .shp in my process.

```
"layers": {
		"water":{ "minzoom": 0,  "maxzoom": 6, "source": "coastline/water_polygons.shp", "simplify_below": 5, "simplify_level": 0.0001, "simplify_ratio": 2},
		"countries":{ "minzoom": 0,  "maxzoom": 7, "source": "countries/ne_10m_admin_0_countries.shp", "simplify_below": 6, "simplify_level": 0.0003, "simplify_ratio": 2 }
	},
```

### Part 2 : Rendering

Once generated, it will need to be "appended" to any specific area generated with the traditional process of tilemaker. The second issue I'm facing at the moment, is that since this basemap  (basemap.mbtiles) will logically be less detailed than the specific area (area.mbtiles), when both are rendered and served, I can still "see" the basemap layer under the more detailed area (for instance the border at higher zoom will not match anymore between both mbtiles). Is there any way to avoid this behavior ? My workaround at the moment is to render the basemap up to z6 (the highest zoom used to generate details) but that doesn't work if I'm zooming on an area that is not covered within the area.mbtiles, since in that case the server does not have any other tiles to display (hope that makes sense) ?

I'm using maplibre-gl and tileserver-gl

Really appreciate any feedback!

**Comment by systemed** [comment]
I think what you want to do is make sure your area.mbtiles doesn't include any coastline/ocean or country border data. Then make sure your simplify settings result in sufficiently detailed coastline at z13/14 but significantly simplified at lower zooms. Here's mine:

````
"ocean"   : { 
    "minzoom": 2 , "maxzoom": 14,
    "source": "/usr/local/share/maps/shapefiles/water-polygons-split-4326/water_polygons.shp",
    "filter_below": 13, "filter_area": 0.5, "simplify_below": 13, "simplify_level": 0.0005
}
````

**Comment by BenOnTrack** [comment]
So basically, your basemap only contains the layer "ocean" (with the above config file) which, based on the `water-polygons`, will display the coastline and ocean ? 
Is there anything to configure for the --bbox (is -180,-90,180,90 relevant for the basemap) ?

Hence any area generated on top should not include this .shp, nor any border data. 
I'm going to try and will keep you posted!

**Reply by BenOnTrack** [reply]
I might be doing something wrong but with the following configuration, the mbtiles would contain 217098738 tiles. 

```
{
	"layers": {
		"water": {
			"minzoom": 0,
			"maxzoom": 14,
			"source": "coastline/water_polygons.shp",
			"filter_below": 13,
			"filter_area": 0.5,
			"simplify_below": 13,
			"simplify_level": 0.0005
		}
	},
	"settings": {
		"minzoom": 0,
		"maxzoom": 14,
		"basezoom": 14,
		"include_ids": false,
		"combine_below": 14,
		"name": "Tilemaker to OpenMapTiles schema",
		"version": "3.0",
		"description": "Tile config based on OpenMapTiles schema",
		"compress": "gzip",
		"filemetadata": {
			"tilejson": "2.0.0",
			"scheme": "xyz",
			"type": "baselayer",
			"format": "pbf",
			"tiles": ["https://example.com/liechtenstein/{z}/{x}/{y}.pbf"]
		}
	}
}
```

After a few hours, I've stopped the process and the file was already more than 5gb...

**Reply by systemed** [reply]
What’s the command line?

**Comment by BenOnTrack** [comment]
`./tilemaker --output coastline.mbtiles --bbox -180,-90,180,90 --process process-coastline.lua --config config-coastline.json`

I'm wondering if the bbox should be adjusted ?

**Reply by BenOnTrack** [reply]
@systemed did you get any chance to look into that ? 
Is it a bbox issue ? or am I supposed to expect a x.Gb basemap mbtiles for the entire world ?

**Reply by systemed** [reply]
On holiday so not yet. But yes, a basemap for the whole world will be big. Missing out polar regions can make a surprising difference.
